server/src/main/java/com/vaadin/data/BeanPropertySet.java

```diff
@@ -313,7 +313,7 @@ private BeanPropertySet(InstanceKey<T> instanceKey) {
                     .map(descriptor -> new BeanPropertyDefinition<>(this,
                             instanceKey.type, descriptor))
                     .collect(Collectors.toMap(PropertyDefinition::getName,
-                            Function.identity()));
+                            Function.identity(), (pd1, pd2) -> pd2));
```

Comment:
I'd document here that it's okay to pick either pd1 or pd2 (it's okay to pick any of the getter Method both from interface A or B) since the reflection call will ultimately call the same getter method on an implementor of B, thanks to `getProperty()` being virtual.